protocol/x/clob/e2e/order_removal_test.go

```diff
@@ -361,6 +365,362 @@ func TestConditionalOrderRemoval(t *testing.T) {
+		// 		OperationsQueue: []clobtypes.OperationRaw{
+		// 			clobtestutils.NewOrderRemovalOperationRaw(
+		// 				constants.LongTermOrder_Carl_Num0_Id2_Clob0_Buy10_Price35_GTB20_RO.OrderId,
+		// 				clobtypes.OrderRemoval_REMOVAL_REASON_INVALID_REDUCE_ONLY,
+		// 			),
+		// 		},
+		// 	},
+		// 	expectedErr: "Order fill must increase position size or change side",
+		// },
+		"invalid proposal: non reduce-only order may not be removed with reduce-only reason": {
+			subaccounts: []satypes.Subaccount{
+				constants.Carl_Num0_1BTC_Short,
+			},
+			orders: []clobtypes.Order{
+				constants.LongTermOrder_Carl_Num0_Id0_Clob0_Buy1BTC_Price50000_GTBT10,
+			},
+			msgProposedOperations: &clobtypes.MsgProposedOperations{
+				OperationsQueue: []clobtypes.OperationRaw{
+					clobtestutils.NewOrderRemovalOperationRaw(
+						constants.LongTermOrder_Carl_Num0_Id0_Clob0_Buy1BTC_Price50000_GTBT10.OrderId,
+						clobtypes.OrderRemoval_REMOVAL_REASON_INVALID_REDUCE_ONLY,
+					),
+				},
+			},
+			expectedErrType: clobtypes.ErrInvalidOrderRemoval,
+			expectedErr:     "Order must be reduce only",
+		},
+		"invalid proposal: conditional fok order cannot be removed when untriggered": {
+			subaccounts: []satypes.Subaccount{
+				constants.Carl_Num0_10000USD,
+			},
+			orders: []clobtypes.Order{
+				constants.ConditionalOrder_Carl_Num0_Id0_Clob0_Buy05BTC_Price50000_GTBT10_SL_50003_FOK,
+			},
+			msgProposedOperations: &clobtypes.MsgProposedOperations{
+				OperationsQueue: []clobtypes.OperationRaw{
+					clobtestutils.NewOrderRemovalOperationRaw(
+						constants.ConditionalOrder_Carl_Num0_Id0_Clob0_Buy05BTC_Price50000_GTBT10_SL_50003_FOK.OrderId,
+						clobtypes.OrderRemoval_REMOVAL_REASON_CONDITIONAL_FOK_COULD_NOT_BE_FULLY_FILLED,
+					),
+				},
+			},
+			expectedErrType: clobtypes.ErrStatefulOrderDoesNotExist,
+			expectedErr:     "does not exist in triggered conditional state",
+		},
+		"invalid proposal: conditional fok order removal is for non fok order": {
+			subaccounts: []satypes.Subaccount{
+				constants.Carl_Num0_10000USD,
+			},
+			orders: []clobtypes.Order{
+				constants.ConditionalOrder_Carl_Num0_Id0_Clob0_Buy05BTC_Price50000_GTBT10_SL_50003_IOC,
+			},
+			priceUpdate: &prices.MsgUpdateMarketPrices{
+				MarketPriceUpdates: []*prices.MsgUpdateMarketPrices_MarketPrice{
+					prices.NewMarketPriceUpdate(0, 5_000_400_000),
+				},
+			},
+			msgProposedOperations: &clobtypes.MsgProposedOperations{
+				OperationsQueue: []clobtypes.OperationRaw{
+					clobtestutils.NewOrderRemovalOperationRaw(
+						constants.ConditionalOrder_Carl_Num0_Id0_Clob0_Buy05BTC_Price50000_GTBT10_SL_50003_IOC.OrderId,
+						clobtypes.OrderRemoval_REMOVAL_REASON_CONDITIONAL_FOK_COULD_NOT_BE_FULLY_FILLED,
+					),
+				},
+			},
+			expectedErrType: clobtypes.ErrInvalidOrderRemoval,
+			expectedErr:     "Order is not fill-or-kill",
+		},
+		"invalid proposal: conditional fok order cannot be removed when fully filled": {
```

Comment:
not sure if we need a case for partially fill since it's technically invalid

Reply:
there isn't a test for partial fill here, this is testing to ensure we cannot include an Order Removal for a conditional fok order that was fully-filled. The order removal reason is to remove fok orders which were not fully filled, but in this test the order is fully-filled, so it throws an error